Frameworks/BlockchainCertificates/Requests/CertificateValidationRequest.swift

```diff
@@ -281,58 +315,143 @@ public class CertificateValidationRequest : CommonRequest {
+                    return
+                }
+            }
+            
             self?.state = .checkingRevokedStatus
         }
         request.resume()
     }
     
     internal func checkRevokedStatus() {
-        let batchRevoked : Bool = (revokedAddresses?.contains(self.revokationKey!))!
-        if batchRevoked {
-            self.state = .failure(reason: "Certificate Batch has been revoked by issuer. Revocation key is \(self.revokationKey!)")
-            return
-        }
-        if self.certificate.recipient.revocationAddress != nil {
-            let certificateRevoked : Bool = (revokedAddresses?.contains(self.certificate.recipient.revocationAddress!))!
-            if certificateRevoked {
-                self.state = .failure(reason: "Certificate has been revoked by issuer. Revocation key is \(self.certificate.recipient.revocationAddress!)")
+        if certificate.version == .two {
+            guard let url = certificate.issuer.revocationURL else {
+                // Issuer does not revoke certificates
+                // Success
+                self.state = .success
+                return
+            }
+            let request = session.dataTask(with: url) { [weak self] (data, response, error) in
+                guard let response = response as? HTTPURLResponse,
+                    response.statusCode == 200 else {
+                        self?.state = .failure(reason: "Got invalid response from \(url)")
+                        return
+                }
+                guard let data = data else {
+                    self?.state = .failure(reason: "Got a valid response, but no data from \(url)")
+                    return
+                }
+                guard let json = try? JSONSerialization.jsonObject(with: data) as! [String: AnyObject] else {
+                    self?.state = .failure(reason: "Certificate didn't return valid JSON data from \(url)")
+                    return
+                }
+                
+                guard let revokedAssertions = json["revokedAssertions"] as? Array<[String: String]> else {
+                    self?.state = .failure(reason: "Couldn't parse revoked assertions")
+                    return
+                }
+                
+                for ra in revokedAssertions {
+                    let id = ra["id"]
+                    let reason = ra["revocationReason"]
+                    if id == self?.certificate.assertion.uid {
+                        self?.state = .failure(reason: "Certificate has been revoked by issuer. Revoked assertion uid  is \(id!) and reason is \(reason!)")
+                        return
+                    }
+                }
+            
+                // Success
+                self?.state = .success
+            }
+            request.resume()
+        } else {
+            let batchRevoked : Bool = (revokedAddresses?.contains(self.revocationKey!))!
```

Comment:
These `!`s worry me...although it appears that it was already in my code! So, bad past-me. :)